tests/suites/test_suite_aes.function

```diff
@@ -18,12 +18,15 @@ void aes_encrypt_ecb( data_t * key_str, data_t * src_str,
     mbedtls_aes_init( &ctx );
 
 
-    TEST_ASSERT( mbedtls_aes_setkey_enc( &ctx, key_str->x, key_str->len * 8 ) == setkey_result );
+    TEST_ASSERT( mbedtls_aes_setkey_enc( &ctx, key_str->x, key_str->len * 8 )
+                                         == setkey_result );
     if( setkey_result == 0 )
     {
-        TEST_ASSERT( mbedtls_aes_crypt_ecb( &ctx, MBEDTLS_AES_ENCRYPT, src_str->x, output ) == 0 );
+        TEST_ASSERT( mbedtls_aes_crypt_ecb( &ctx, MBEDTLS_AES_ENCRYPT,
+                     src_str->x, output ) == 0 );
 
-        TEST_ASSERT( hexcmp( output, hex_dst_string->x, 16, hex_dst_string->len ) == 0 );
+        TEST_ASSERT( memcmp( output, hex_dst_string->x, hex_dst_string->len )
```

Comment:
`memcmp` alone is not enough. It is used on the assumption that supplied length is greater of the two buffers' lengths and on exceeding the shorter buffer there would be a mismatch of data. But if the supplied length is shorter then the comparison can erroneously pass. Example:
```
A: 123456789101112
B: 123456789
memcmp(A, B, 9)   will pass. Though there could be an error in the test data or output is unexpected!
```

Reply:
There can be a `macro` that does a length comparison before invoking `memcmp`.

Reply:
Or two TEST_ASSERTs, like:
```C
TEST_ASSERT( hex_dst_string->len == 16 );
TEST_ASSERT( memcmp( output, hex_dst_string->x, hex_dst_string->len );
```

Reply:
A one liner will be more maintainable.
```
TEST_ASSERT_BUFFER_EQUAL(b1, b1_len, b2, b2_len) ...
```